src/goto-programs/label_function_pointer_call_sites.cpp

```diff
@@ -52,17 +52,31 @@ void label_function_pointer_call_sites(goto_modelt &goto_model)
           goto_model.symbol_table.lookup_ref(call_site_symbol_name)
             .symbol_expr();
 
-        // add assignment to the new function pointer variable, followed by a
-        // call of the new variable
+        // add a DECL instruction for the function pointer variable
+        auto decl_instruction =
+          goto_programt::make_decl(new_function_pointer, source_location);
+        goto_function.second.body.insert_before_swap(it, decl_instruction);
+        it++;
```

Comment:
Nit picking: `++it` (which the compiler will likely take care of for you, but why rely on compiler optimisations...).